File: getBkn/bkn.js
function getBkn(skey) {
    let hashValue = 5381;
    for (let i = 0; i < skey.length; i++) {
      hashValue += (hashValue << 5) + skey.charCodeAt(i);
    }
    return hashValue & 2147483647;
  }

File: getBkn/bkn.java
public class Main {
    public static int getBkn(String skey) {
        int hashValue = 5381;
        for (char c : skey.toCharArray()) {
            hashValue += (hashValue << 5) + (int) c;
        }
        return hashValue & 2147483647;
    }
}